bundles/org.openhab.binding.fmiweather/src/main/java/org/openhab/binding/fmiweather/internal/client/FMIResponse.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.fmiweather.internal.client;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * Class representing response from the FMI weather service
+ *
+ * @author Sami Salonen - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class FMIResponse {
+
+    /**
+     * Builder class for FMIResponse
+     *
+     * @author Sami Salonen
+     *         author Sami Salonen - Initial contribution/
+     *
+     */
+    public static class Builder {
+
+        private Map<Location, Map<String, List<Long>>> timestampsByLocationByParameter;
+        private Map<Location, Map<String, List<@Nullable BigDecimal>>> valuesByLocationByParameter;
+
+        public Builder() {
+            timestampsByLocationByParameter = new HashMap<>();
+            valuesByLocationByParameter = new HashMap<>();
+        }
+
+        public Builder appendLocationData(Location location, @Nullable Integer capacityHintForValues, String parameter,
+                long epochSecond, @Nullable BigDecimal val) {
+            timestampsByLocationByParameter.computeIfAbsent(location, k -> new HashMap<>()).computeIfAbsent(parameter,
+                    k -> capacityHintForValues == null ? new ArrayList<>() : new ArrayList<>(capacityHintForValues))
+                    .add(epochSecond);
+            valuesByLocationByParameter.computeIfAbsent(location, k -> new HashMap<>()).computeIfAbsent(parameter,
+                    k -> capacityHintForValues == null ? new ArrayList<>() : new ArrayList<>(capacityHintForValues))
+                    .add(val);
+            return this;
+        }
+
+        public FMIResponse build() {
+            Map<Location, Map<String, Data>> out = new HashMap<>(timestampsByLocationByParameter.size());
+
+            timestampsByLocationByParameter.entrySet().stream().forEach(entry -> {
+                Location location = entry.getKey();
+                Map<String, List<Long>> timestampsByParameter = entry.getValue();
+                out.put(location, new HashMap<String, Data>(timestampsByParameter.size()));
+                timestampsByParameter.entrySet().stream().forEach(entry2 -> {
+                    String parameter = entry2.getKey();
+                    long[] timestamps = entry2.getValue().stream().mapToLong(Long::longValue).toArray();
+                    @Nullable
```

Comment:
Does this annotation has any effect?

Reply:
64433ad662b8ac8928f6641f73b8c7cd26bf2ddc